metrics-influxdb/src/main/java/io/dropwizard/metrics5/influxdb/GarbageFreeScheduledReporter.java

```diff
@@ -0,0 +1,199 @@
+import io.dropwizard.metrics5.Timer;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public abstract class GarbageFreeScheduledReporter extends ScheduledReporter {
+
+    private MetricRegistry registry;
+    private RegistryMirror mirror;
+    
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name, 
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit) {
+        super(registry, name, filter, rateUnit, durationUnit);
+        init(registry, filter);
+    }
+
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name, 
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit, 
+            ScheduledExecutorService executor) {
+        super(registry, name, filter, rateUnit, durationUnit, executor);
+        init(registry, filter);
+    }
+
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name, 
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit, 
+            ScheduledExecutorService executor, 
+            boolean shutdownExecutorOnStop) {
+        super(registry, name, filter, rateUnit, durationUnit, executor, shutdownExecutorOnStop);
+        init(registry, filter);
+    }
+
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name,
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit, 
+            ScheduledExecutorService executor, 
+            boolean shutdownExecutorOnStop, 
+            Set<MetricAttribute> disabledMetricAttributes) {
+        super(registry, name, filter, rateUnit, durationUnit, executor, shutdownExecutorOnStop, disabledMetricAttributes);
+        init(registry, filter);
+    }
+    
+    private void init(MetricRegistry registry, MetricFilter filter) {
+        this.registry = registry;
+        this.mirror = new RegistryMirror(filter);
+        mirror.register(registry);
+    }
+
+    @Override
+    public void stop() {
+        try {
+            super.stop();
+        } finally {
+            mirror.unregister(registry);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("rawtypes")
+    public void report() {
+        synchronized (this) {
+            report(mirror.gauges(), 
+                    mirror.counters(), 
+                    mirror.histograms(), 
+                    mirror.meters(), 
+                    mirror.timers());
+        }
+    }
+    
+    private static class RegistryMirror implements MetricRegistryListener {
+
+        private final MetricFilter filter;
+        @SuppressWarnings("rawtypes") // because of signature in ScheduledReporter#report(..)
```

Comment:
I guess we could be less specific and just to place this annotation over the class.